### ISSUE TITLE
Jemalloc support for Ruby

### ISSUE THREAD

**Issue by mikhailov** [travis-ci/travis-ci#8853]
I realised that Ruby compiled in Travis VMs do not have `jemalloc` support. It will be beneficial for tests perform time and saving resources on Travis side as well.

I used this script to check:
```
ruby -r rbconfig -e "puts RbConfig::CONFIG['LIBS'].match('jemalloc').nil? ? 'Ruby does not have jemalloc support' : 'Ruby compiled with jemalloc'"
```

I'd like to have Ruby built with `jemalloc` via set ENV VAR `RUBY_CONFIGURE_OPTS=--with-jemalloc` before Travis kicks off `rvm use $(< .ruby-version) --install --binary --fuzzy`. Would that be possible to do via .travis.yml configuration file? I can't make it work, tried to have it `before_install` but unfortunatelly it happens after rvm use command described above.

**Comment by BanzaiMan** [comment]
Try rerunning the `rvm` command after setting the environment variable.

**Comment by mikhailov** [comment]
Not a bad idea by adds an extra 30sec to every job to download/compile Ruby (actually by compiling Ruby two times in a row on T2 type of AWS instances it steals CPU credits very quick)... Any other ideas? Thanks!

**Comment by BanzaiMan** [comment]
Override the Ruby version specified by `.ruby-version` so that there is no extra installation.

**Comment by mikhailov** [comment]
Interesting, what do you mean by _override the Ruby version_? I do have Ruby specified in .ruby-version file, where can I override it?

How to make sure the pre-installed version does have jemalloc compiled in? Thank you for your help!

**Comment by BanzaiMan** [comment]
Please refer to https://docs.travis-ci.com/user/languages/ruby

```yaml
language: ruby
rvm: '2.4' # or another Ruby that's pre-installed on the image.
⋮
```

However, if this is a compile-time option, using `rvm` to install it may not yield the fruit you are after.

**Comment by mikhailov** [comment]
Makes sense, so there is no option to define global ENV VARs (to use to compile Ruby) even via Travis UI? 

How about pre-installed Ruby complied with jemalloc?

**Comment by mikhailov** [comment]
@BanzaiMan btw when I set `rvm: '2.3.5'` I still see:
```
$ rvm use 2.3.5 --install --binary --fuzzy
ruby-2.3.5 is not installed - installing.
```

Is there any other steps to take in order to avoid Ruby compiliing? Thank you!

**Comment by cotsog** [comment]
You should be able to compile and install a Ruby runtime with jemalloc using a recipe similar to the following:

```
before_install:
  - sudo apt-get install libjemalloc-dev
  - rvm install 2.3.5 -C --with-jemalloc
  - ruby -r rbconfig -e "puts RbConfig::CONFIG['LIBS']" # this should print `-ljemalloc` to confirm
```

The caveat with this method, however, is that it will take 4 minutes every build just to get this setup. You can add the following recipe to cache the compiled Ruby runtime for the faster installation after the first time:

```
before_cache: pushd ~/.rvm && rm -rf archives rubies/ruby-2.2.7 rubies/ruby-2.3.4 && popd
cache:
  directories: ~/.rvm
```

This should bring down the installation time to ~1 minute.
